src/renderer.js

```diff
@@ -76,78 +76,94 @@ class Renderer {
       // Check that all outstanding network requests have finished loading.
       const outstandingRequests = new Map();
       let initialRequestId = undefined;
+      let networkIdleTimeout = undefined;
+
       Network.requestWillBeSent((event) => {
         if (!initialRequestId)
           initialRequestId = event.requestId;
         outstandingRequests.set(event.requestId, event);
+        if (networkIdleTimeout) {
+          clearTimeout(networkIdleTimeout);
+          networkIdleTimeout = undefined;
+          networkIdle = false;
+        }
       });
 
       let statusCode = 200;
+
       Network.responseReceived((event) => {
-        if (event.requestId == initialRequestId && event.response.status != 0)
+        if (event.requestId == initialRequestId &&
+          event.response.status != 0 &&
+          event.response.status != 304)
```

Comment:
why not 304?

Reply:
There's a browser cache inside the running instance. When we load the same page twice, we may get a 304 because its inside rendertron's browser cache (which we want to leverage). This, however, is not safe to return the originating client.

If we do return a 304 (which was a bug previously), the client would have the incorrect contents. 304s can be returned by rendertron if the status codes are 200, and the etag matches what the requesting client has. This is currently implemented by default when using Express.